modules/mapbox/src/deck-utils.js

```diff
@@ -63,9 +60,10 @@ export function updateLayer(deck, layer) {
 }
 
 export function drawLayer(deck, layer) {
-  // set layerFilter to only allow the current layer
-  deck.deckRenderer.layerFilter = params => shouldDrawLayer(layer.id, params.layer);
-  deck._drawLayers('mapbox-repaint');
+  deck._drawLayers('mapbox-repaint', {
+    layers: getLayers(deck, deckLayer => shouldDrawLayer(layer.id, deckLayer)),
```

Comment:
This is out of scope for this PR, but I would somehow have preferred if we just passed along a `layerFilter` here?

As a general design nit, it feels unnecessarily convoluted that we need to wrap 	`deckRenderer.draw` in a separate `Deck._drawLayers` call. The call stack here is already deep and each layer in this "matryoshka" adds to the "cognitive load" when reading the code.

I wonder if `DeckRenderer.setProps()` could set the layers and viewports and we could just call `deckRenderer.draw({layerFilter})` without the `Deck._drawLayers` wrapper.

Reply:
I don't like this either, though changing it will require nested changes made to DeckRenderer and all the passes. I agree that the `layerFilter` prop handling needs to be revisited.